burger/toppings/version.py

```diff
@@ -92,6 +96,49 @@ def act(aggregate, classloader, verbose=False):
             aggregate["version"]["is_flattened"] = False
             aggregate["version"]["entity_format"] = "1.10"
 
+        if "protocol" in aggregate["version"] and aggregate["version"]["protocol"] > 92:
+            # Although 13w39b (80) was the last version before the rewrite, the highest protocol version belongs to 2.0 Purple (92)
+            # If the current one is any higher, it's guaranteed to be a post netty-rewrite version
+            # This will cover versions 15w51a (93) and onwards
+            aggregate["version"]["netty_rewrite"] = True
+        elif aggregate["version"]["distribution"] == "server":
+            # If it's a server-specific file, we can just look for any netty class
+            # Any version prior to 15w51a (93) is guaranteed to have their dependencies shaded directly on the jar file
+            aggregate["version"]["netty_rewrite"] = "io/netty/buffer/ByteBuf" in classloader.classes
```

Comment:
Ah, wait, this assumption is incorrect: shading stopped at some point in 1.18 and instead they use a bundling system of some sort. But running burger on the server jar doesn't work properly anyways (I guess you'd need to extract the bundled `minecraft_server.23w14a.jar/META-INF/versions/23w14a/server-23w14a.jar` instead, but I haven't tried it), so it's not that important.

Reply:
Oh, really? I assumed it was a normal thing to run it on the server jar, at least prior to the merge, due to missing data on one side. After that, there's really not much reason since the client has everything we need, so I didn't bother to handle +1.18 servers (in fact, they will fall on the "unknown" distribution type, but should still work if you extract the inner version jar from the bundle and run on it).